### ISSUE TITLE
Paths broken for subsection pages?

### ISSUE THREAD

**Issue by phil-opp** [getzola/zola#128]
(I'm not quite sure about this issue. Maybe I'm missing something obvious.)

I just updated [os.phil-opp.com](https://os.phil-opp.com/) to gutenberg 0.2.0 and now the page links in this section are broken: https://os.phil-opp.com/extra/naked-exceptions/

The pages override the `path` in the front matter, but the pages in `section.pages` still have the original path value. So the links point to the normal path, but the pages are placed under the overridden path.

**Example**: The [Catching Exceptions](https://os.phil-opp.com/catching-exceptions/) post has `path = catching-exceptions` [in its front matter](https://github.com/phil-opp/blog_os/blob/7590050ffea472c23b811aab4796256402568ac0/blog/content/extra/naked-exceptions/01-catching-exceptions/index.md). This seems to work, at least the page is placed under that path. However, the page variable in `section.pages` contains the wrong path `extra/naked-exceptions/catching-exceptions`, which leads to a broken link on the [section page](https://os.phil-opp.com/extra/naked-exceptions/) (rendered using these templates: [[1]](https://github.com/phil-opp/blog_os/blob/7590050ffea472c23b811aab4796256402568ac0/blog/templates/handling-exceptions-with-naked-fns.html) [[2]](https://github.com/phil-opp/blog_os/blob/7590050ffea472c23b811aab4796256402568ac0/blog/templates/section.html) [[3]](https://github.com/phil-opp/blog_os/blob/7590050ffea472c23b811aab4796256402568ac0/blog/templates/macros.html)).

**Comment by Keats** [comment]
That's weird, I can't see where the bug could be at first glance.
I'm in the middle of moving continents so I don't have much time this week, it would be super helpful if you can make a failing test in `components/site` for that and I'll have a look tomorrow.

**Comment by phil-opp** [comment]
I found the problem. The `page.path` value is correct, it is just a relative/absolute path problem.

I just made the mistake of linking to the `path` directly. So gutenberg creates a `<a href="catching-exceptions">`, which is a link relative to the current directory. What I really want is an absolute path (`/catching-exceptions`). Is it possible to generate absolute paths in gutenberg?

For now, I'm using `page.permalink`, but this unneccessarily prepends the base URL.
